src/Surfnet/StepupRa/RaBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -69,6 +80,14 @@ public function __construct(
      */
     public function authenticate(TokenInterface $token)
     {
+        if ($token->assertion->getIssuer() !== $this->remoteIdp) {
```

Comment:
As discussed IRL, it is simpler to verify the configured remote IdP is the same as the Issuer of the response in `SamlInteractionProvider`'s `processSamlResponse`. If ever we get more than one remote IdP and thus need state, it can be refactored to something different.